src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -402,48 +404,66 @@ FindNodeCheckInRangeTableList(List *rtable, bool (*check)(Node *))
 
 
 /*
- * QueryContainsDistributedTableRTE determines whether the given
+ * QueryContainsDistributedNonReferenceTableRTE determines whether the given
  * query contains a distributed table.
  */
 bool
-QueryContainsDistributedTableRTE(Query *query)
+QueryContainsDistributedNonReferenceTableRTE(Query *query)
 {
 	return FindNodeCheck((Node *) query, IsDistributedTableRTE);
```

Comment:
Did you intend IsDistributedNonReferenceTableRTE?

Reply:
Yes